Content.Shared/Stealth/SharedStealthSystem.cs

```diff
@@ -54,10 +58,30 @@ public virtual void SetEnabled(EntityUid uid, bool value, StealthComponent? comp
         if (!Resolve(uid, ref component, false) || component.Enabled == value)
             return;
 
+        if (!_mobState.IsDead(uid) && value == false)
+            return;
+
         component.Enabled = value;
         Dirty(component);
     }
 
+    private void OnMobStateChanged(EntityUid uid, StealthComponent component, MobStateChangedEvent args)
+    {
+        if (args.NewMobState != MobState.Dead)
+        {
+            if (!component.Enabled)
+            {
+                component.Enabled = true;
+                Dirty(component);
+            }
+        }
+        else
+        {
+            component.Enabled = false;
+            Dirty(component);
+        }
+    }
```

Comment:
Those dirty calls are obsolete, you should use the other one.

You should also just be calling SetEnabled here instead and not just disabling stealth toggling on mobs.

Reply:
Done. 
But how to replace the obsolete Dirty, I have no idea. It should be done by someone else or a more skilful coder's help is needed.

Reply:
Dirty(uid, component);

Reply:
Yeah. It's done. Thanks for your help.